### PR TITLE
Register user

### PR DESCRIPTION
**Pull request by NguyenThiHangB**

### CODE REVIEW

**Comment by NguyenThiHangB** [comment]
screenshot from 2018-07-25 15-34-45

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      flash[:danger] = t ".danger"
+      render :new
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:username, :email, :password,
+      :password_confirmation)
```
Comment (mdmahmudurrahman) [inline app/controllers/users_controller.rb]: Remove redundant ()

File: app/views/shared/_error_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t ".error" %> <%= pluralize(object.errors.count, t(".pluralize")) %>.
```
Comment (mdmahmudurrahman) [inline app/views/shared/_error_messages.html.erb]: 1. Remove redundant ()

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<% provide :title, t(".title") %>
+<% provide :button_text, t(".button") %>
+<h1><%= t(".title") %></h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @user do |f| %>
+    <%= render "shared/error_messages", object: @user%>
```
Comment (mdmahmudurrahman) [inline app/views/users/new.html.erb]: Add a white space before closing erb symbol

File: config/routes.rb [hunk 4]
```diff
@@ -1,2 +1,3 @@
 Rails.application.routes.draw do
+  resources :users
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Specify action/actions using only

**Pull request merged**